AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp

```diff
@@ -171,7 +171,7 @@ int WebSocketClient::sendLength(long len, char *msg)
         {
             // Actually, we only support message size less than 2^32
             int shift = i * 8 > 32 ? 32 : i * 8;
-            msg[i + 1] = (len >> shift) & 0xff;

```

Comment:
msg[i + 1] = (((unsigned long)len) >> shift) & 0xff;

Reply:
After updated the code, the issue also can repro.

Reply:
long is 32-bit, not 64-bit, so the code should be as below
         msg[0] = 127 | (1 << 7);
         msg[1] = 0;
         msg[2] = 0;
         msg[3] = 0;
         msg[4] = 0;
         msg[5] = (len >> 24) & 0xff;
         msg[6] = (len >> 16) & 0xff;
         msg[7] = (len >> 8) & 0xff;
         msg[8] = len & 0xff;
         return 9;